test/deployframework/helpers.go

```diff
@@ -119,12 +118,6 @@ func (pw *PodWaiter) WaitForPods(namespace string, targetPodsCount int) error {
 			return false, err
 		}
 
-		// TODO(chancez): is this check needed? If so, maybe move outside of
-		// WaitForPods.
-		if len(pods.Items) == 0 {
```

Comment:
We definitely don't need this check anymore, especially with the introduction of running the e2e suite locally (as the metering-operator pod would be run locally.)